interceptors/validator/options.go

```diff
@@ -0,0 +1,69 @@
+package validator
+
+import "github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/logging"
+
+var (
+	defaultOptions = &options{
+		logger:         DefaultLoggerMethod,
+		shouldFailFast: DefaultDeciderMethod,
+	}
+)
+
+type options struct {
+	logger         Logger
+	shouldFailFast Decider
+}
+
+// Option
+type Option func(*options)
+
+func evaluateServerOpt(opts []Option) *options {
+	optCopy := &options{}
+	*optCopy = *defaultOptions
+	for _, o := range opts {
+		o(optCopy)
+	}
+	return optCopy
+}
+
+func evaluateClientOpt(opts []Option) *options {
+	optCopy := &options{}
+	*optCopy = *defaultOptions
+	for _, o := range opts {
+		o(optCopy)
+	}
+	return optCopy
+}
+
+// Logger
+type Logger func() (logging.Level, logging.Logger)
+
+// DefaultLoggerMethod
+func DefaultLoggerMethod() (logging.Level, logging.Logger) {
+	return "", nil
+}
+
+// WithLogger
+func WithLogger(logger Logger) Option {
```

Comment:
Why not here:

```suggestion
func WithLogger(logging.Level logger Logger) Option {
```

Simpler to use (:

Reply:
oops, yaa i might have overthink it!!

Reply:
made the changes made on the suggestions.